allauth/account/forms.py

```diff
@@ -363,7 +355,6 @@ class AddEmailForm(UserForm):
 
     def clean_email(self):
         value = self.cleaned_data["email"]
-        value = get_adapter().clean_email(value)
```

Comment:
I think it will become messy if the email unique handling is made part of the `clean_email`, as the `clean_email` method is unaware of the use case (signup, add-email, password reset). You would need to pass along an additional parameter to indicate whether or not the email is going to be added, or just used for input cleaning. Therefore, I think it is best to keep the original `clean_email` as is, and add a dedicated method for the uniqueness checking: `validate_email_uniqueness`?